sentry-android/src/main/java/io/sentry/android/AndroidSerializer.java

```diff
@@ -1,17 +1,37 @@
 package io.sentry.android;
 
+import com.google.gson.FieldNamingPolicy;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
 import io.sentry.ISerializer;
 import io.sentry.SentryEnvelope;
 import io.sentry.SentryEvent;
+import io.sentry.protocol.SentryId;
+import java.util.Date;
 
 public class AndroidSerializer implements ISerializer {
+
+  private Gson gson =
+      new GsonBuilder()
+          .setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES)
+          .registerTypeAdapter(SentryId.class, new SentryIdSerializerAdapter())
+          .registerTypeAdapter(SentryId.class, new SentryIdDeserializerAdapter())
+          .registerTypeAdapter(Date.class, new DateSerializerAdapter())
+          .registerTypeAdapter(Date.class, new DateDeserializerAdapter())
+          .create();
+
   @Override
   public SentryEnvelope deserializeEnvelope(String envelope) {
-    return null;
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public SentryEvent deserializeEvent(String envelope) {
+    return gson.fromJson(envelope, SentryEvent.class);
```

Comment:
Do we ever deserialize in the client?

Reply:
We will for communication between sentry-android and sentry-native.

Reply:
because of NDK, yes